src/library/scala/collection/mutable/TreeMap.scala

```diff
@@ -162,5 +162,7 @@ sealed class TreeMap[A, B] private (tree: RB.Tree[A, B])(implicit val ordering:
       iterator.foreach { case (key, value) => update(key, f(key, value)) }
       this
     }
+
+    override def clone() = super.clone().rangeImpl(from, until)
```

Comment:
This is related to the set/tree view semantics I mentioned below. While writing this implementation of `TreeSet` I noticed an issue in `clone()` that also affected `mutable.TreeMapView`'s `clone()` (the two collections share practically the same implementation, only with different interfaces).

Consider this:

```scala
scala> val map = mutable.TreeMap(3 -> 4, 4 -> 5, 5 -> 6)
map: scala.collection.mutable.TreeMap[Int,Int] = TreeMap(3 -> 4, 4 -> 5, 5 -> 6)

scala> val view = map.from(4) # `view` is an instance of TreeMapView
view: scala.collection.mutable.TreeMap[Int,Int] = TreeMap(4 -> 5, 5 -> 6)

scala> val view2 = view.clone() # `view2` is an instance of TreeMap!
view2: scala.collection.mutable.TreeMap[Int,Int] = TreeMap(4 -> 5, 5 -> 6)

scala> view += 1 -> 1
res0: view.type = TreeMap(4 -> 5, 5 -> 6)

scala> view2 += 1 -> 1
res1: view2.type = TreeMap(1 -> 1, 4 -> 5, 5 -> 6)
```

If `clone()` is not overriden in views, it will yield a normal `TreeSet`/`TreeMap` and the previous view bounds are lost.